src/dndbeyond/base/utils.js

```diff
@@ -292,10 +296,17 @@ async function buildAttackRoll(character, attack_source, name, description, prop
             if (no_type_idx > -1) {
                 damage_types[no_type_idx] = "Chaotic Energy";
             }
-        } else if (roll_properties.name == "Toll the Dead") {
-            const ttd_dice = await dndbeyondDiceRoller.queryGeneric(roll_properties.name, "Is the target missing any of its hit points ?", { "d12": "Yes", "d8": "No" }, "ttd_dice", ["d12", "d8"]);
-            if (ttd_dice === null) return null;
-            damages[0] = damages[0].replace("d8", ttd_dice);
+        } else if (roll_properties.name.includes("Toll the Dead")) {
+            if (character.getSetting("toll-choice") === "both") {
+                damages.splice(1, 0, upgradeDice(damages[0], 8, 12));
+                damage_types.splice(1, 0, damage_types[0] + ' (Missing HP)');
+                damage_types[0] = damage_types[0] + ' (Full HP)';
+            } else {
+                const ttd_dice = await dndbeyondDiceRoller.queryGeneric(roll_properties.name, "Is the target missing any of its hit points ?", { "d12": "Yes", "d8": "No" }, "ttd_dice", ["d12", "d8"]);
+                if (ttd_dice === null) return null;
+                damages[0] = damages[0].replace("d8", ttd_dice);
```

Comment:
Could replace this with `upgradeDice` too (or use the same `.replace`. I don't think we need the additional `upgradeDice` method for this simple replacement)